### PR TITLE
[FEAT]: Add user profile retrieval tests with role-based access control

### PR DESCRIPTION
**Pull request by shresthashim**
Issue : https://github.com/OpsiMate/OpsiMate/issues/364

# Add API Tests for Get User Profile Endpoint

## Changes

- Added comprehensive API tests for `GET /api/v1/users/profile` endpoint
- Tests cover authentication, authorization, response structure, and edge cases
- Validates users can only access their own profile data securely

## Test Scenarios

✅ Valid authenticated requests (200 OK)
❌ Unauthenticated access (401 Unauthorized)
🔒 Different user roles can access their profiles
🧾 Response structure validation (excludes sensitive data)

## Summary by CodeRabbit

* **Tests**
  * Expanded test suite for the user profile endpoint to validate secure profile retrieval with authentication requirements, role-based access control, proper rejection of unauthorized access attempts, data integrity verification, and correct error handling for missing users.

### CODE REVIEW

File: apps/server/tests/users.test.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import request, { SuperTest, Test } from 'supertest';
-import Database from 'better-sqlite3';
-import { createApp } from '../src/app.js';
-import { Role } from '@OpsiMate/shared';
+import request, { SuperTest, Test } from "supertest";
```
Comment (yuvalmich) [inline apps/server/tests/users.test.ts:1]: Please remove not needed changes like this

Comment (idanlodzki) [reply apps/server/tests/users.test.ts:1]: @shresthashim - Please remove not needed changes like this

File: apps/server/tests/users.test.ts [hunk 2]
```diff
@@ -250,11 +250,7 @@ describe("Users API", () => {
 
       // Check that all users are present
       const emails = res.body.data.map((u: any) => u.email).sort();
-      expect(emails).toEqual([
-        "admin@example.com",
-        "editor@example.com",
-        "viewer@example.com",
-      ]);
+      expect(emails).toEqual(["admin@example.com", "editor@example.com", "viewer@example.com"]);
```
[the remark this answers is not in the document]
Comment (idanlodzki) [reply apps/server/tests/users.test.ts:253]: @shresthashim please fix
